scripts/run-json2ts.js

```diff
@@ -102,7 +102,9 @@ Promise.all(schemaFiles.map(schemaFile => {
   // prepend 'JSON' to all identifiers here so they don't conflict with main .ts types
   for (let i = 0; i < dedupedDefs.length; ++i) {
     for (let id of added) {
-      dedupedDefs[i] = dedupedDefs[i].replace(new RegExp(`\\b${id}\\b`), id + 'JSON');
+      // 1) To avoid FooJSON: FooJSON instead of a more consistent Foo: FooJSON we don't replace if followed by :
+      // 2) For some reason we're getting recursive types having a 1 appended to the type so cover that too
```

Comment:
Did we ever figure out the "For some reason" here?

Reply:
No but I didn't look into their implementation or anything. We had problems even getting recursive types to work in the first place so maybe this is some workaround the json schema-related tools we call do to get around it, but it's been a long time so I don't remember the specifics. It's an issue for tx metadata + plutus datums as those are our only recursive types.